src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -550,6 +550,29 @@ public void EmptyEnumStoreTest()
             Assert.True(enums.Count() > 0);
         }
 
+        [Fact]
+        public async void MaxRecursionDepthTest()
+        {
+            bool test;
+            var config = new PowerFxConfig(null)
+            {
+                MaxCallDepth = 5
+            };
+            var recalcEngine = new RecalcEngine(config);
+            try
+            {
+                await recalcEngine.EvalAsync("Abs(Abs(Abs(Abs(Abs(Abs(1))))))", new System.Threading.CancellationToken());
+                test = false;
+            }
+            catch (MaxCallDepthException)
+            {
+                test = true;
+            }
+
+            Assert.True(test);
+            await recalcEngine.EvalAsync("Abs(Abs(Abs(Abs(Abs(1)))))", new System.Threading.CancellationToken());
+        }

```

Comment:
Can you try this test:

Sum(
  Sum(Sum(1),1),
  Sum(Sum(1),1),
  Sum(Sum(1),1),
)

Reply:
It passes, if I remove the trailing comma.

Reply:
Second one fails with
Message: 
System.InvalidOperationException : Errors: Error 20-21: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 41-42: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 62-63: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 83-84: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 104-105: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 125-126: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 126-126: Expected an operand. The formula or expression expects a valid operand. For example, you can add the operand '2' to the expression ' 1 +_' so that the result is '3'. Or, you can add the operand "there" to the expression '"Hi "& _ ' so that the result is 'Hi there'.
Error 125-126: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 104-105: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 83-84: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 62-63: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 41-42: Expected operator. We expect an operator such as +, *, or & at this point in the formula.
Error 20-21: Expected operator. We expect an operator such as +, *, or & at this point in the formula.

Reply:
Probably needs EnableExpressionChaining flag

Reply:
Just used Sum for test.